modules/govuk/manifests/apps/router_api.pp

```diff
@@ -79,4 +84,25 @@
       value   => join($router_nodes, ','),
     }
   }
+
+  # Set up a cron job which outputs the current nodes for a specific machine class.
+  # The file can then be read in by router-api to publish routes for those nodes.
+  if $::aws_migration {
+    if $router_nodes_class {
+      $router_port = '3055'
+      $router_nodes_file = '/etc/router_nodes'
+
+      cron::crondotdee { 'update-router-nodes':
+        command => "/usr/local/bin/govuk_node_list -c ${router_nodes_class} | sed 's/$/:${router_port}/g' > ${router_nodes_file}",
+        hour    => '*',
+        minute  => '*/5',
+        mailto  => '""',
+      }
+
+      govuk::app::envvar { "${title}-ROUTER_NODES_FILE":
```

Comment:
Does the application get refreshed if this file changes?

Reply:
The application reads in the location of the file rather than the values of the file, and then reads the file to gather the values each time it needs to reload router, so it _should_ read in the newest values (at least that's the theory).

Reply:
I don't think its going to work like that yet. The nodes are stored in a variable associated with the class [1], which is set when the Rails initialisers run.

The RouterReloader is used from the controller here [2], and it just calls the reload function. The reload function then reads the value in `router_reload_urls`.

1: https://github.com/alphagov/router-api/blob/b1be434b7e55e587ecdb8f2e4165eecb41293120/lib/router_reloader.rb#L9
2: https://github.com/alphagov/router-api/blob/master/app/controllers/routes_controller.rb#L41